src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -907,6 +907,76 @@ public override IOperation VisitWhileLoop(IWhileLoopOperation operation, int? ca
             return null;
         }
 
+        public override IOperation VisitVariableDeclarationGroup(IVariableDeclarationGroupOperation operation, int? captureIdForResult)
+        {
+            // We erase variable declarations from the control flow graph, as variable lifetime information is contained in a parallel data structure.
+            foreach (var declaration in operation.Declarations)
+            {
+                HandleVariableDeclaration(declaration, captureIdForResult);
+            }
+
+            return null;
+        }
+
+        private void HandleVariableDeclaration(IVariableDeclarationOperation operation, int? captureIdForResult)
+        {
+            foreach (IVariableDeclaratorOperation declarator in operation.Declarators)
+            {
+                ILocalSymbol localSymbol = declarator.Symbol;
+
+                IOperation initializer = null;
+                SyntaxNode assignmentSyntax = null;
+                if (declarator.Initializer != null)
+                {
+                    initializer = Visit(declarator.Initializer.Value, captureIdForResult);
+                    assignmentSyntax = declarator.Syntax;
+                }
+
+                if (operation.Initializer != null)
+                {
+                    IOperation operationInitializer = Visit(operation.Initializer.Value, captureIdForResult);
```

Comment:
Will we visit the `operation.Initializer` multiple times here for a declaration with multiple declarators sharing the same initializer? I feel we should visit them only once, and re-use the visited result for all assignments (probably move `operationInitializer` declaration outside the loop, and execute the visit only if `operationInitializer == null`?)

Reply:
We cannot reuse the node as it should have a different parent in each assignment

---
In reply to: [166980240](https://github.com/dotnet/roslyn/pull/24692#discussion_r166980240) [](ancestors = 166980240)

Reply:
Also, Value can contain control flow, which will add statements and basic blocks, simply reusing the returned node will not add these "side-effects".

---
In reply to: [167017796](https://github.com/dotnet/roslyn/pull/24692#discussion_r167017796) [](ancestors = 167017796,166980240)

Reply:
@333fred Please add a PROTOTYPE comment that we should add a test with a control flow in a shared initializer. We can add it right now, but it I think it won't produce expected result because we don't handle control flow in object creation yet. 

---
In reply to: [167020856](https://github.com/dotnet/roslyn/pull/24692#discussion_r167020856) [](ancestors = 167020856,167017796,166980240)

Reply:
Added.

---
In reply to: [167023541](https://github.com/dotnet/roslyn/pull/24692#discussion_r167023541) [](ancestors = 167023541,167020856,167017796,166980240)